lib/std/os/windows.zig

```diff
@@ -601,25 +602,50 @@ pub fn GetCurrentDirectory(buffer: []u8) GetCurrentDirectoryError![]u8 {
     return buffer[0..end_index];
 }
 
-pub const CreateSymbolicLinkError = error{Unexpected};
+pub const CreateSymbolicLinkError = error{ AccessDenied, PathAlreadyExists, FileNotFound, Unexpected };
 
 pub fn CreateSymbolicLink(
     sym_link_path: []const u8,
     target_path: []const u8,
-    flags: DWORD,
+    is_directory: bool,
 ) CreateSymbolicLinkError!void {
     const sym_link_path_w = try sliceToPrefixedFileW(sym_link_path);
     const target_path_w = try sliceToPrefixedFileW(target_path);
-    return CreateSymbolicLinkW(sym_link_path_w.span().ptr, target_path_w.span().ptr, flags);
+    return CreateSymbolicLinkW(sym_link_path_w.span().ptr, target_path_w.span().ptr, is_directory);
 }
 
 pub fn CreateSymbolicLinkW(
     sym_link_path: [*:0]const u16,
     target_path: [*:0]const u16,
-    flags: DWORD,
+    is_directory: bool,
 ) CreateSymbolicLinkError!void {
-    if (kernel32.CreateSymbolicLinkW(sym_link_path, target_path, flags) == 0) {
+    // Previously, until Win 10 Creators Update, creating symbolic links required
+    // SeCreateSymbolicLink privilege. Currently, this is no longer required if the
+    // OS is in Developer Mode; however, SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE
+    // must be added to the input flags.
+    const flags = if (is_directory) SYMBOLIC_LINK_FLAG_DIRECTORY else 0;
+    if (kernel32.CreateSymbolicLinkW(sym_link_path, target_path, flags | SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE) == 0) {
```

Comment:
Keep in mind that the plan forward is #1840, and so this code that builds around `kernel32.CreateSymbolicLinkW` is climbing up a local maximum rather than a global maximum. In other words, this code will eventually have to be rewritten to use NtDll rather than kernel32. Maybe it would be worth investing in converting status quo to NtDll and making progress on that front?

Reply:
Sounds good! Lemme try and massage this a bit further and see what I can come up with!

Reply:
FWIW, if I manage to weed out all uses of winapi, I'll be able to get rid of the confusing prefixing helpers `sliceToWin32PrefixedFileW` which will no longer be needed (I hope!).

Reply:
Done!